Assets/Scripts/System/SceneJsonReader.cs

```diff
@@ -0,0 +1,209 @@
+using Assets.Scripts.EditorState;
+using Assets.Scripts.SceneState;
+using Assets.Scripts.System;
+using Newtonsoft.Json;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using UnityEngine;
+using UnityEngine.InputSystem.Controls;
+using Zenject;
+using static Assets.Scripts.System.SceneLoadSaveSystem;
+
+public class SceneJsonReader
+{
+    private static DecentralandSceneData decentralandSceneData;
+
+    // Dependencies
+    private IPathState _pathState;
+
+    [Inject]
+    private void Construct(IPathState pathState)
+    {
+        _pathState = pathState;
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    /// <summary>
+    /// Loads the Scene Data that the user works on. Loads it just once and stores it as a static variable
+    /// </summary>
+    /// <param name="reload">if it should reload the scene data from the json file again</param>
+    /// <returns>the Decentraland scene data; if there is an error it returns null</returns>
+    public DecentralandSceneData getSceneData(bool reload)
+    {
+        try
+        {
+            var json = File.ReadAllText(_pathState.ProjectPath + "/scene.json");
```

Comment:
You should move this inside the `if`. When we don't want/need to reload the file, we also don't want to read it